engine/runtime-compiler/src/main/java/org/enso/compiler/phase/exports/ExportSymbolAnalysis.java

```diff
@@ -0,0 +1,270 @@
+package org.enso.compiler.phase.exports;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.enso.compiler.PackageRepository;
+import org.enso.compiler.context.CompilerContext;
+import org.enso.compiler.core.ir.Diagnostic;
+import org.enso.compiler.core.ir.DiagnosticStorage;
+import org.enso.compiler.core.ir.MetadataStorage;
+import org.enso.compiler.core.ir.Module;
+import org.enso.compiler.core.ir.Name;
+import org.enso.compiler.core.ir.expression.errors.ImportExport;
+import org.enso.compiler.core.ir.module.scope.Export;
+import org.enso.compiler.data.BindingsMap.ResolvedType;
+import org.enso.compiler.pass.IRPass;
+import scala.collection.immutable.Seq;
+import scala.collection.immutable.Seq$;
+import scala.jdk.javaapi.CollectionConverters;
+
+/**
+ * Ensures that all the symbols that are exported exist. If not, an IR error is generated. It is not
+ * a {@link IRPass compiler pass} because it needs access to {@link
+ * org.enso.compiler.PackageRepository}. Ignores renamed exports.
```

Comment:
I don't understand the explanation why this isn't a compiler pass.

I mean I'm not convinced it has to be.

But we _can_ access `PackageRepository` in passes, I've been doing this for #9812.

```java
var packageRepo = moduleContext.pkgRepo().isDefined() ? moduleContext.pkgRepo().get() : null;
```

or is something wrong about this pattern?